java/src/main/c++/vw_VW.cc

```diff
@@ -76,6 +76,34 @@ JNIEXPORT jfloat JNICALL Java_vw_VW_predict(JNIEnv *env, jobject obj, jstring ex
     return prediction;
 }
 
+JNIEXPORT jfloatArray JNICALL Java_vw_VW_multipredict(JNIEnv *env, jobject obj, jstring example_string, jlong vwPtr) {
+  jfloatArray r = env->NewFloatArray(0);
+  try {
+    vw* vwInstance = (vw*)vwPtr;
+    const char *utf_string = env->GetStringUTFChars(example_string, NULL);
+    example *ec = VW::read_example(*vwInstance, utf_string);
+
+    vwInstance->l->predict(*ec);
+
+    r = env->NewFloatArray(vwInstance->lda);
+    float *predictions = ec->topic_predictions.begin;
+    jfloat *weights = new jfloat[vwInstance->lda];
+    for (size_t k = 0; k < vwInstance->lda; k++) {
+      weights[k] = predictions[k];
+    }
+    env->SetFloatArrayRegion(r, 0, vwInstance->lda, weights);
+    delete weights;
+
+    ec->in_use = true; // temporary workaround for issue #800
```

Comment:
@jmorra should this workaround be committed as part of this PR or leave it out and wait for a fix separately with issue #800?